R/FilterStateChoices.R

```diff
@@ -195,6 +201,12 @@ ChoicesFilterState <- R6::R6Class( # nolint
       do.call(super$initialize, args)
 
       private$set_choices(choices)
+
+      if (is.null(selected) && multiple) {
+        selected <- private$choices
+      } else if (is.null(selected)) {
+        selected <- private$choices[1]
+      }
```

Comment:
There is potential problem on init:
```r
ChoicesFilterState$new(x = letters, dataname = "data", varname = "var", selected = c("b", "c"), multiple = FALSE)
```

```suggestion
      } else if (length(selected) > 1 && !multiple) {
        warning(
          "FilterStateChoices allows selected to be length=1 when multiple is FALSE. ",
          "Only first value is taken."
        )
        selected <- selected[1]
      }
```

Please include this test after change
```r
testthat::test_that("ChoicesFilterState constructor forces single selected when multiple is FALSE", {
  testthat::expect_warning(
    state <- ChoicesFilterState$new(
      x = letters, dataname = "data", varname = "var", selected = c("b", "c"), multiple = FALSE
    )
   )
   testthat::expect_identical(
     shiny::isolate(state$get_state()$selected),
     "b"
   )
})
```